### PR TITLE
Upgrade to Storybook 5

### PR DESCRIPTION
**Pull request by kalleericson**
Screenshot 2019-04-11 at 11 11 24

# Storybook 5
- Strictly upgrade, with minimal optimization
- Project open to restyle; optimize and implement new features

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -4,7 +4,7 @@ stories/
 coverage
 lib/examples/
 .gitignore
-prettier.config.js
+prettier.config.ts
```
Comment (lindskogen) [inline .npmignore]: This change is not correct, right?

File: examples/grid-hooks/GridExample.tsx [hunk 2]
```diff
@@ -9,6 +8,8 @@ import { DefaultText } from '../../src/components/ui/text/DefaultText';
 import { useGridCell } from '../../src/features/grid-hooks/hooks/UseGridCell';
 import { createIndexArray, createRows, FocusedBox } from './util/GridHooksExampleUtils';
 
+const { useCallback, useState } = React;
```
Comment (lindskogen) [inline examples/grid-hooks/GridExample.tsx]: Revert this

File: examples/grid-hooks/GridExampleWithContext.tsx [hunk 3]
```diff
@@ -14,6 +13,8 @@ import {
   FocusedBox,
 } from './util/GridHooksExampleUtils';
 
+const { useCallback, useState } = React;
```
Comment (lindskogen) [inline examples/grid-hooks/GridExampleWithContext.tsx]: Revert this

File: package.json [hunk 4]
```diff
@@ -8,6 +8,7 @@
   "types": "lib/index",
   "scripts": {
     "start": "start-storybook -p 6006",
+    "start-strong": "NODE_OPTIONS=\"--max-old-space-size=8192\" start-storybook -p 6006",
```
Comment (lindskogen) [inline package.json]: Revert this

File: src/components/ui/form/date-time-input/TimeTextInput.tsx [hunk 5]
```diff
@@ -1,12 +1,12 @@
 import { faClock } from '@fortawesome/free-regular-svg-icons/faClock';
 import * as React from 'react';
-import { useCallback, useState } from 'react';
 import { formatTimeString, validUserInput } from '../../../../util/time';
 import { useTheme } from '../../../theme/UseThemeHook';
 import {
   DefaultTextInput,
   DefaultTextInputProps,
 } from '../text-input/DefaultTextInput';
+const { useCallback, useState } = React;
```
Comment (lindskogen) [inline src/components/ui/form/date-time-input/TimeTextInput.tsx]: Revert this

File: src/components/ui/form/switch/Switch.tsx [hunk 6]
```diff
@@ -13,6 +12,8 @@ import { Icon } from '../../icon';
 import { ValueOnChangeProps } from '../types';
 import { SwitchTheme } from './SwitchTheme';
 
+const { useCallback } = React;
```
Comment (lindskogen) [inline src/components/ui/form/switch/Switch.tsx]: Revert this

File: src/components/ui/layout/Box.tsx [hunk 7]
```diff
@@ -3,7 +3,7 @@ import styled from '@emotion/styled';
 import {
   alignItems,
   AlignItemsProps,
-  bgColor,
+  // bgColor,
```
Comment (lindskogen) [inline src/components/ui/layout/Box.tsx]: Remove comments

File: src/components/ui/layout/Box.tsx [hunk 8]
```diff
@@ -51,13 +50,13 @@ export interface BoxProps extends StyledSystemProps {
   row?: boolean;
   spacing?: boolean | number;
   indent?: boolean | number;
-  style?: CSSProperties;
+  style?: React.CSSProperties;
 }
-
+// bgColor
```
Comment (lindskogen) [inline src/components/ui/layout/Box.tsx]: Remove comment?

File: src/components/ui/popup/__tests__/EnhancedPopper.test.tsx [hunk 9]
```diff
@@ -12,7 +12,6 @@ it('works', () => {
 // import { shallow } from 'enzyme';
 // import { createSerializer } from 'jest-emotion';
 // import * as React from 'react';
-// import { ReactElement } from 'react';
```
Comment (lindskogen) [inline src/components/ui/popup/__tests__/EnhancedPopper.test.tsx]: Revert this

File: src/components/ui/popup/__tests__/Popup.test.tsx [hunk 10]
```diff
@@ -9,7 +9,6 @@ it('works', () => {
 });
 // import { shallow } from 'enzyme';
 // import * as React from 'react';
-// import { ReactElement } from 'react';
```
Comment (lindskogen) [inline src/components/ui/popup/__tests__/Popup.test.tsx]: Revert this

File: src/features/grid-hooks/hooks/UseEditableCell.ts [hunk 11]
```diff
@@ -1,11 +1,6 @@
-import {
-  KeyboardEvent,
-  KeyboardEventHandler,
-  useCallback,
-  useMemo,
-  useState,
-} from 'react';
+import * as React from 'react';
 import { RevertableValue, useRevertableValue } from './UseRevertableValue';
+const { useCallback, useMemo, useState } = React;
```
Comment (lindskogen) [inline src/features/grid-hooks/hooks/UseEditableCell.ts]: Revert this

File: .npmignore (hunk 1 above)
Comment (kalleericson) [reply .npmignore]: Haha wuuut? How the f did this happen?!

File: src/components/ui/layout/Box.tsx [hunk 12]
```diff
@@ -51,13 +49,12 @@ export interface BoxProps extends StyledSystemProps {
...
   indent?: boolean | number;
-  style?: CSSProperties;
+  style?: React.CSSProperties;
 }
-
 const FlexBox = styled.div<FlexBoxProps>`
   display: ${props => props.display || 'flex'};
   ${alignItems}
-  ${bgColor}
+  background-color: #fafafa;
```
[the remark this answers is not in the document]
Comment (kalleericson) [reply src/components/ui/layout/Box.tsx]: Nice catch, fixed

**Pull request merged**